src/log.ts

```diff
@@ -886,27 +887,33 @@ class Log implements LogSeverityFunctions {
       const payloadSize = JSON.stringify(entry).length;
       if (payloadSize < this.maxEntrySize) return;
 
-      const delta = payloadSize - this.maxEntrySize;
+      let delta = payloadSize - this.maxEntrySize;
       if (entry.textPayload) {
         entry.textPayload = entry.textPayload.slice(
           0,
           Math.max(entry.textPayload.length - delta, 0)
         );
       } else {
-        // Stackdriver Log Viewer picks up the summary line from the
-        // 'message' field.
-        if (
-          entry.jsonPayload &&
-          entry.jsonPayload.fields &&
-          entry.jsonPayload.fields.message &&
-          entry.jsonPayload.fields.message.stringValue
-        ) {
-          const text: string | null | undefined =
-            entry.jsonPayload.fields.message.stringValue;
-          entry.jsonPayload.fields.message.stringValue = text.slice(
-            0,
-            Math.max(text.length - delta, 0)
-          );
+        const fieldsToTruncate = [
+          // Winston:
+          'jsonPayload.fields.metadata.structValue.fields.stack.stringValue',
+          // Bunyan:
+          'jsonPayload.fields.msg.stringValue',
+          'jsonPayload.fields.err.structValue.fields.stack.stringValue',
+          'jsonPayload.fields.err.structValue.fields.message.stringValue',
+          // All:
+          'jsonPayload.fields.message.stringValue',
+        ];
+        for (const field of fieldsToTruncate) {
+          const msg: string = dotProp.get(entry, field, '');
+          if (msg !== '') {
```

Comment:
Is it possible that `msg` will be undefined? I'm just wondering if we should add a `typeof` check in addition to this.

Reply:
no, the third argument to `dotProp.get` is the default, so it will default to `''` in the case that there's nothing at that path.

Reply:
boop: https://www.npmjs.com/package/dot-prop#getobject-path-defaultvalue